### PR TITLE
Update tape definitions to 5.6

### PR DESCRIPTION
**Pull request by r-thomson**
See this PR for previous discussion: https://github.com/DefinitelyTyped/DefinitelyTyped/pull/65537

I basically went through what was already here and compared it with the README to make sure everything looks right. I also converted some of the comments to JSDOC and reduced the duplication for method aliases.

---

Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: <<url here>>
- [x] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.

### CODE REVIEW

**Comment by DangerBotOSS** [comment]
Inspecting the JavaScript source for this package found some properties that are not in the .d.ts files.
The check for missing properties isn't always right, so take this list as advice, not a requirement.
## tape ([<kbd>unpkg</kbd>](https://unpkg.com/browse/tape@latest/))
was missing the following properties:
1. wait
1. run
1. Test
1. test
<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against 793aa1de95ea6568291aec0f67c7230135430cf1
</p>

File: types/tape/index.d.ts [hunk 1]
```diff
@@ -139,94 +157,101 @@ declare namespace tape {
          * Assert that value is truthy with an optional description message msg.
          */
         ok(value: any, msg?: string, extra?: AssertOptions): void;
-        true(value: any, msg?: string, extra?: AssertOptions): void;
-        assert(value: any, msg?: string, extra?: AssertOptions): void;
+        true: Test['ok'];
+        assert: Test['ok'];
```
Comment (ljharb) [inline types/tape/index.d.ts:161]: nice :-)

**Review by ljharb — APPROVED** [review]
Seem great, thanks!

**Comment by ljharb** [comment]
Ready to merge

**Pull request merged**